### PR TITLE
ci: move install_yq function in lib.sh a dedicate script

### PR DESCRIPTION
**Pull request by marcov**
Moving install_yq in a dedicated script allows other repositories to reuse it.
Also, this complements the set of scripts in .ci/ used to install software.

Fixes: #1604

Signed-off-by: Marco Vedovati <mvedovati@suse.com>

### CODE REVIEW

File: .ci/install_yq.sh [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+function install_yq() {
+	local yq_path="${GOPATH}/bin/yq"
+	local yq_pkg="github.com/mikefarah/yq"
+	[ -x  "${GOPATH}/bin/yq" ] && return
+
+	read -r -a sysInfo <<< "$(uname -sm)"
+
+	case "${sysInfo[0]}" in
+	"Linux" | "Darwin")
```
Comment (grahamwhaley) [inline .ci/install_yq.sh]: nit: I think we tend to indent the switch items in the rest of our bash scripts ;-)

File: .ci/install_yq.sh [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+		goos="${sysInfo[0],}"
+		;;
+	"*")
+		die "OS ${sysInfo[0]} not supported"
+		;;
+	esac
+
+	case "${sysInfo[1]}" in
+	"aarch64")
```
Comment (grahamwhaley) [inline .ci/install_yq.sh]: hmm, I thought we had a func to do the arch name conversions - but, I see this is the code we originally had in the `install_yq` func, so just a note, not a blocker.

File: .ci/install_yq.sh [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+#
+# Copyright (c) 2019 SUSE LLC
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+set -e
+
+function install_yq() {
+	local yq_path="${GOPATH}/bin/yq"
```
Comment (grahamwhaley) [inline .ci/install_yq.sh]: I wonder if we should check `GOPATH` is set?
It is a long standing out of date requirement of our CI, as `go` no longer needs `GOPATH` to be set and will default.

Comment (jodh-intel) [reply .ci/install_yq.sh]: Yep - you could just source `.ci/lib.sh` which handles this:

- https://github.com/kata-containers/tests/blob/master/.ci/lib.sh#L22..L28

Comment (marcov) [reply .ci/install_yq.sh]: I can add a `set -u` to fail if GOPATH is not set. I think for the time being `$GOPATH/bin` is the most obvious destination for the `yq` executable.

**Comment by marcov** [comment]
Thanks @grahamwhaley for the review, I think they all makes sense. I force-pushed some updates following your suggestions.

File: .ci/install_yq.sh (hunk 3 above)
Comment (marcov) [reply .ci/install_yq.sh]: missed you comment @jodh-intel. I am fine doing that too. It's going to introduce some mutual dependency between the two file, but that shouldn't be circular.

**Comment by marcov** [comment]
@jodh-intel @grahamwhaley @devimc I figured out some things on Travis with macOS were broken, so I booted up my Mac and fixed those. Can you take another look please?

/test

**Pull request merged**